internal/pkg/mount/mount_test.go

```diff
@@ -4,11 +4,123 @@
 
 package mount_test
 
-import "testing"
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
 
-func TestEmpty(t *testing.T) {
-	// added for accurate coverage estimation
-	//
-	// please remove it once any unit-test is added
-	// for this package
+	"github.com/stretchr/testify/suite"
+	"github.com/talos-systems/go-blockdevice/blockdevice/loopback"
+	"golang.org/x/sys/unix"
+
+	"github.com/talos-systems/talos/internal/pkg/mount"
+	"github.com/talos-systems/talos/pkg/makefs"
+)
+
+// Some tests in this package cannot be run under buildkit, as buildkit doesn't propagate partition devices
+// like /dev/loopXpY into the sandbox. To run the tests on your local computer, do the following:
+//
+//  go test -exec sudo -v --count 1 github.com/talos-systems/talos/internal/pkg/mount
+
+type manifestSuite struct {
+	suite.Suite
+
+	disk           *os.File
+	loopbackDevice *os.File
+}
+
+const (
+	diskSize = 4 * 1024 * 1024 * 1024 // 4 GiB
+)
+
+func TestManifestSuite(t *testing.T) {
+	suite.Run(t, new(manifestSuite))
+}
+
+func (suite *manifestSuite) SetupTest() {
+	suite.skipIfNotRoot()
+
+	var err error
+
+	suite.disk, err = ioutil.TempFile("", "talos")
+	suite.Require().NoError(err)
+
+	suite.Require().NoError(suite.disk.Truncate(diskSize))
+
+	suite.loopbackDevice, err = loopback.NextLoopDevice()
+	suite.Require().NoError(err)
+
+	suite.T().Logf("Using %s", suite.loopbackDevice.Name())
+
+	suite.Require().NoError(loopback.Loop(suite.loopbackDevice, suite.disk))
+
+	suite.Require().NoError(loopback.LoopSetReadWrite(suite.loopbackDevice))
+}
+
+func (suite *manifestSuite) TearDownTest() {
+	if suite.loopbackDevice != nil {
+		suite.Assert().NoError(loopback.Unloop(suite.loopbackDevice))
+	}
+
+	if suite.disk != nil {
+		suite.Assert().NoError(os.Remove(suite.disk.Name()))
+		suite.Assert().NoError(suite.disk.Close())
+	}
+}
+
+func (suite *manifestSuite) skipUnderBuildkit() {
+	hostname, _ := os.Hostname() //nolint:errcheck
+
+	if hostname == "buildkitsandbox" {
+		suite.T().Skip("test not supported under buildkit as partition devices are not propagated from /dev")
+	}
+}
+
+func (suite *manifestSuite) skipIfNotRoot() {
+	if os.Getuid() != 0 {
+		suite.T().Skip("can't run the test as non-root")
+	}
+}
+
+func (suite *manifestSuite) TestCleanCorrupedXFSFileSystem() {
+	suite.skipUnderBuildkit()
```

Comment:
I believe this test should run under `buildkit`, as it's using raw block device without partitions!